std-bits/table/src/main/java/org/enso/table/data/column/builder/InferredBuilder.java

```diff
@@ -160,11 +167,25 @@ private void retypeAndAppend(Object o) {
       }
     }
 
+    if (allowDateToDateTimeConversion
+        && o instanceof LocalDate
+        && currentBuilder.canRetypeTo(DateTimeType.INSTANCE)) {
+      currentBuilder = currentBuilder.retypeTo(DateTimeType.INSTANCE);
+      currentBuilder.append(o);
+      return;
+    }
```

Comment:
I'm not a fan of this special case - the whole concept of `RetypeInfo` pairs was to encode the possible changes of types. Any new logic should firstly be added through that instead of adding special `if`s.

Anyway, this special case should no longer be needed, right? It should be handled by `DateBuilder` with the `allowDateToDateTimeConversion` set, right?

Reply:
Unless it's impossible to do this through RetypeInfo, a special case is acceptable I think. But I don't see why it couldn't work using the 'normal' approach?